### PR TITLE
LDAP: fix getDomainInfo() to set this as di.backend

### PR DESCRIPTION
**Pull request by goestreicher**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by zeha** [comment]
Note that getDomainInfo may be called for domains that are not stored in the backend. In this case, the DomainInfo object must NOT be changed.

**Comment by goestreicher** [comment]
@zeha Yes, and although the diff doesn't show it, this case is already considered (the DomainInfo object will only be called if there's a SOARecord)

**Comment by zeha** [comment]
@goestreicher Ah, the diff was just too confusing.

**Comment by goestreicher** [comment]
Is there something preventing this from being merged? Just ping me if that's the case :)

**Comment by Habbie** [comment]
We where in strong release mode the last few weeks. Now that we have released, and created maintenance branches, git master is back in "just do it" mode. I'll see what I can do during the coming holidays!

File: modules/ldapbackend/ldapbackend.cc [hunk 1]
```diff
@@ -697,7 +695,7 @@ bool LdapBackend::getDomainInfo( const DNSName& domain, DomainInfo& di )
     // search for SOARecord of domain
     filter = "(&(associatedDomain=" + toLower( m_pldap->escape( domain.toStringRootDot() ) ) + ")(SOARecord=*))";
     m_msgid = m_pldap->search( getArg( "basedn" ), LDAP_SCOPE_SUBTREE, filter, attronly );
-    m_pldap->getSearchEntry( msgid, result );
```
Comment (rgacogne) [inline modules/ldapbackend/ldapbackend.cc]: Just a nit but it looks like `msgid` is now unused so we should just remove it.

Comment (aerique) [reply modules/ldapbackend/ldapbackend.cc]: @goestreicher Could you please address the nit mentioned above? Then I'll merge it.

**Comment by aerique** [comment]
Gah, meant to press cancel to edit the merge commit but accidentally pressed the green button!

**Pull request merged**